mysql-test/suite/rpl/t/rpl_show_slave_hosts.test

```diff
@@ -49,4 +49,497 @@ source include/wait_show_condition.inc;
+
+--let $save_default_master_timeout= `SELECT @@GLOBAL.rpl_semi_sync_master_timeout`
+select @@global.rpl_semi_sync_master_timeout;
+show variables like 'rpl_semi_sync_master_enabled';
+
+# Master status must be `ON`
+let $status_var= rpl_semi_sync_master_status;
+let $status_var_value= ON;
+source include/wait_for_status_var.inc;
+show status like 'rpl_semi_sync_master_status';
+
+show status like 'rpl_semi_sync_master_get_ack';
+set global rpl_semi_sync_master_timeout=0;
+select @@global.rpl_semi_sync_master_timeout;
+# here rpl_semi_sync_master_status=0 means async is taking place (no wait on ACK from slave)
+show variables like 'rpl_semi_sync_master_timeout';
+show status like 'rpl_semi_sync_master_status';
+
+connection slave;
+--source include/stop_slave.inc
+--let $save_server_2_dbug= `SELECT @@GLOBAL.DEBUG_DBUG`
+SET @@GLOBAL.DEBUG_DBUG="d,synchronize_semisync_slave_reply";
+--source include/start_slave.inc
+let $status_var= rpl_semi_sync_slave_status;
+let $status_var_value= ON;
+source include/wait_for_status_var.inc;
+show status like 'rpl_semi_sync_slave%';
+SET debug_sync='now WAIT_FOR at_slave_reply TIMEOUT 10';
+
+connection slave2;
+--source include/stop_slave.inc
+--let $save_server_3_dbug= `SELECT @@GLOBAL.DEBUG_DBUG`
+SET @@GLOBAL.DEBUG_DBUG="d,synchronize_semisync_slave_reply";
+--source include/start_slave.inc
+let $status_var= rpl_semi_sync_slave_status;
+let $status_var_value= ON;
+source include/wait_for_status_var.inc;
+show status like 'rpl_semi_sync_slave%';
+SET debug_sync='now WAIT_FOR at_slave_reply TIMEOUT 10';
+
+--echo # Both replicas are stopped with synchronization point
+connection master;
+# Write event
+INSERT INTO t values (1111);
+# There will be no ACK so good enough to check if the same rpl_semi_sync_master_get_ack is observed
+let $status_var= rpl_semi_sync_master_get_ack;
+let $status_var_value= 9;
+source include/wait_for_status_var.inc;
+show status like 'rpl_semi_sync_master_get_ack';
+
+# GTID_sent > GTID_ack (ACK is empty)
+--replace_column 3 'SLAVE_PORT'
+--replace_result $SLAVE_MYPORT SLAVE_PORT $DEFAULT_MASTER_PORT DEFAULT_PORT
+SHOW REPLICA HOSTS;
+
+connection slave;
+SET debug_sync='now SIGNAL reply_ack_to_master';
+
+--echo # Replica server_2 have sent ACK
+connection master;
+# GTID_sent == GTID_ack (updated to the latest active transaction)
+# This is weird, master is async, why would it "get" ACK?
+let $status_var= rpl_semi_sync_master_get_ack;
+let $status_var_value= 10;
+source include/wait_for_status_var.inc;
+show status like 'rpl_semi_sync_master_get_ack';
+--replace_column 3 'SLAVE_PORT'
+--replace_result $SLAVE_MYPORT SLAVE_PORT $DEFAULT_MASTER_PORT DEFAULT_PORT
+SHOW REPLICA HOSTS;
+
+connection slave2;
+SET debug_sync='now SIGNAL reply_ack_to_master';
+
+--echo # Replica server_3 sent ACK (it shouldn't increment ACK to master?)
+connection master;
+# GTID_sent == GTID_ack (already sent by server_id=2) - this is not good, ack master should be 9 instead of 10?
```

Comment:
This will increase for each ACK received; not for each transaction that needs an ACK; so 11 here is correct.